sale_cart_delivery_rest_api/services/cart.py

```diff
@@ -107,7 +107,9 @@ def _set_delivery_method(self, cart, params):
             raise UserError(_("This delivery method is not available for you order"))
         cart.write({"carrier_id": method_id})
         if self._must_charge_delivery_fee_on_order():
-            cart.delivery_set()
+            cart.set_delivery_line(
+                cart.carrier_id, cart.carrier_id.rate_shipment(cart)["price"]
+            )
```

Comment:
Please be particularly careful when reviewing this part as I am not comfortable with delivery carriers, thus I deduced (maybe wrongly) how to migrate this line.